controllers/users.js

```diff
@@ -64,15 +65,58 @@ exports.modifyUser = async (req, res) => {
     email: req.body.email,
   });
 
-  await User.updateOne({ _id: req.params.id }, userToModify).then(() => {
-    res.status(200).json({
-      message: 'User updated successfully!',
-    });
-  }).catch(
-    (error) => {
+  await User.updateOne({ _id: req.params.id }, userToModify)
+    .then(() => {
+      res.status(200).json({
+        message: 'User updated successfully!',
+      });
+    })
+    .catch((error) => {
       res.status(400).json({
         error,
       });
-    },
-  );
+    });
+};
+
+exports.modifyPassword = async (req, res) => {
+  const user = await User.findOne({ _id: req.params.id });
+
+  async function passwordValidation() {
+    try {
+      const matchPassword = await bcrypt.compare(
+        req.body.password,
+        user.password,
+      );
+      if (!matchPassword) {
+        return res.status(401).json({ error: 'Mot de passe incorrect !' });
+      }
+      return true;
+    } catch (error) {
+      res.status(500).json({ error });
+    }
+
+    return undefined;
+  }
+
+  const isValidPassword = await passwordValidation();
+
+  if (isValidPassword === true) {
+    try {
+      const hashedPassword = await bcrypt.hash(req.body.newPassword, 10);
+
+      const passwordToUpdate = new User({
+        _id: req.params.id,
+        password: hashedPassword,
+      });
+
+      await User.updateOne({ _id: req.params.id }, passwordToUpdate);
+      return res
+        .status(200)
+        .json({ message: 'Mot de passe a été mis a jour!' });
+    } catch (error) {
+      return res.status(400).json({ error });
+    }
+  }
+
+  return undefined;
```

Comment:
do we need to return undefined here? Or isn't it better to return an updated password?